docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,1494 @@
+        direction LR
+        NodeC(Node C)
+    end
+    subgraph AS 65001 [AS 65001: External Network]
+        direction LR
+        ToR1(ToR Switch 1)
+        ToR2(ToR Switch 2)
+    end
+    NodeA <-- eBGP --> ToR1
+    NodeB <-- eBGP --> ToR1
+    NodeC <-- eBGP --> ToR2
+    ToR1  <-- iBGP --> ToR2
+```
+
+##### Full-Mesh iBGP across the Cluster
+
+The FRR instance on each node maintains a full-mesh BGP peer relationship with
+all other nodes across the cluster. In this mode, users can enable no-overlay
+mode without relying on external BGP routes. Enabling `outboundSNAT` for the
+no-overlay network allows pod egress traffic to be SNATed to the node IP,
+enabling communication with external destinations even when the pod network is
+not advertised externally.
+
+This is particularly useful in smaller clusters or in environments where
+external BGP speakers are not available. However, it is important to note that a
+full-mesh iBGP setup can lead to increased CPU and memory consumption on the
+nodes, as each node must maintain a session with every other node.
+
+```mermaid
+graph LR
+    subgraph AS 64512 [ AS 64512 ]
+      direction LR
+      A(Node A)
+      B(Node B)
+      C(Node C)
+      A <-- iBGP --> B
+      B <-- iBGP --> C
+      C <-- iBGP --> A
+    end
+```
+
+Here's an example FRRConfiguration:
+
+```yaml
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  labels:
+    network: default
+  name: receive-all
+  namespace: frr-k8s-system
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      # All nodes are added as neighbors to setup a full-mesh BGP topology
+      neighbors:
+      - address: 192.168.111.20
+        asn: 64512
+        disableMP: true
+        toReceive:
+          allowed:
+            mode: all
+      - address: 192.168.111.21
+        asn: 64512
+        disableMP: true
+        toReceive:
+          allowed:
+            mode: all
+      - address: 192.168.111.22
+        asn: 64512
+        disableMP: true
+        toReceive:
+          allowed:
+            mode: all
+```
+
+##### External Route Reflectors for Larger Clusters
+
+In a large cluster, a full-mesh BGP setup leads to more CPU and memory
+consumption on the nodes. Instead of every node peering with every other node,
+nodes peer only with external BGP route reflectors. This significantly reduces
+the number of BGP sessions each node needs to maintain, improving scalability.
+
+```mermaid
+graph TD
+    subgraph AS 64512 [ AS 64512 ]
+        direction LR
+        subgraph "Route Reflector"
+            RR1(Route Reflector 1)
+        end
+        subgraph "Kubernetes Nodes"
+            NodeA(Node A)
+            NodeB(Node B)
+            NodeC(Node C)
+        end
+    end
+    NodeA <-- iBGP --> RR1
+    NodeB <-- iBGP --> RR1
+    NodeC <-- iBGP --> RR1
+```
+
+### Feature Compatibility
+
+#### Multiple External Gateways (MEG)
+
+MEG cannot work with advertised networks, therefore MEG cannot work with
```

Comment:
hmm I forgot I guess that we excluded MEG support with CDN + BGP. I guess you can ignore my earlier comments about MEG and we can handle feature support in a later OKEP.